### ISSUE TITLE
[BUG] [nvim-cmp] unable to pass a function in sources entryFilter

### ISSUE THREAD

**Issue by AleXoundOS** [nix-community/nixvim#1173]
| Field        | Description |
|--------------|-------------|
| Plugin       | nvim-cmp    | 
| Nixpkgs      | [master](https://github.com/NixOS/nixpkgs/commit/a41cc628170e0c3d42f250c7a8be4d0d41a78c82) | 
| Home Manager | - | 

- [x] I have read the [FAQ](https://nix-community.github.io/nixvim/user-guide/faq.html) and my bug is not listed there.

## Minimal, Reproducible Example (MRE)

```
nvim.nvim-cmp = {
  enable = true;
  sources = [
    {
      name = "nvim_lsp";
      entryFilter = ''
        function(entry, ctx)
          return require('cmp.types').lsp.CompletionItemKind[entry:get_kind()] ~= 'Function'
        end
      '';
    }
  ];
}
```

The `entryFilter` code gets passed as a string into init.lua. Hence, the code doesn't work.

Maybe the problem is related to how https://github.com/nix-community/nixvim/blob/main/plugins/completion/nvim-cmp/setup-options.nix treats the `entryFilter` option. It seems all sources fields are converted to strings unconditionally here: https://github.com/nix-community/nixvim/blob/be87309e0c1da19d99d969625300aaed36ff1b91/plugins/completion/nvim-cmp/setup-options.nix#L68.

---

_Sorry, I couldn't prepare an easy to run Nix flake. I should update the issue later._

**Comment by AleXoundOS** [comment]
Most likely this issue is a sibling of #1157 (which is about LSP keymappings).

**Comment by GaetanLepage** [comment]
This should be fixed by https://github.com/nix-community/nixvim/pull/1174. Let's keep this issue open until it is confirmed to work.

**Comment by GaetanLepage** [comment]
Fixed by #1174
